# Phase 8 Model Evaluation

In [4]:
import pandas as pd

phase4_results = pd.read_csv("comparison\phase4_baseline_results.csv")
phase4_results

<>:2: SyntaxWarning: "\p" is an invalid escape sequence. Such sequences will not work in the future. Did you mean "\\p"? A raw string is also an option.
<>:2: SyntaxWarning: "\p" is an invalid escape sequence. Such sequences will not work in the future. Did you mean "\\p"? A raw string is also an option.
C:\Users\SURYA\AppData\Local\Temp\ipykernel_23452\3263319627.py:2: SyntaxWarning: "\p" is an invalid escape sequence. Such sequences will not work in the future. Did you mean "\\p"? A raw string is also an option.
  phase4_results = pd.read_csv("comparison\phase4_baseline_results.csv")


,Model,MAE,RMSE,MAPE,R2,Bias
0,Naive(1Hr),1054.538423,1354.455901,3.426626,0.955795,-0.203314
1,Naive(24Hr),2206.332124,3027.407171,7.013357,0.779157,-9.834695
2,Moving Average,3558.373439,4468.461840,11.718028,0.518875,-6.741366
3,Previous Day,2206.332124,3027.407171,7.013357,0.779157,-9.834695
4,Previous Week,3434.415810,4735.138844,10.698904,0.459735,-41.495501


In [5]:
phase5_results = pd.read_csv("comparison/phase5_all_results.csv")
phase5_results

,Rank,Model,Input_Hours,Output_Hours,MAE,RMSE,MAPE,R2,Bias
0,1,RNN,24,24,1376.715652,1926.890466,4.347443,0.910540,78.356911
1,2,GRU,48,24,1408.347366,2007.912543,4.424472,0.902920,130.560385
2,3,RNN,48,24,1427.283400,1991.721145,4.540931,0.904480,291.563654
3,4,GRU,24,24,1429.704553,2086.472267,4.503000,0.895100,241.550000
4,5,RNN,168,24,1430.213322,1981.727949,4.582685,0.905324,375.384036
5,6,GRU,168,24,1452.675728,2075.132046,4.598523,0.896189,284.291680
6,7,LSTM,48,24,1480.836807,2166.632235,4.637197,0.886966,244.603077
7,8,LSTM,24,24,1491.632725,2165.564133,4.693110,0.887006,315.432527
8,9,Bi-LSTM,24,24,1532.420000,2220.760000,4.829000,0.881200,225.460000
9,10,LSTM,168,24,1539.109104,2171.032286,4.912044,0.886372,447.831487


In [10]:
phase6_results = pd.read_csv("comparison/final_9_models_comparison.csv")
phase6_results

,Model,Architecture,Input Hours,Output Hours,MAE,RMSE,MAPE (%),R²,Bias
0,RNN + RMSprop + Learning Rate,RNN,168,24,1391.473640,1955.137632,4.431960,0.907848,260.425411
1,RNN Baseline,RNN,24,24,1376.715652,1926.890466,4.347443,0.910540,78.356911
2,RNN + RMSprop,RNN,48,24,1368.041997,1924.233401,4.316486,0.910843,30.372446
3,LSTM 32 Neurons,LSTM,48,24,1400.047234,1995.070825,4.411724,0.904158,89.953964
4,LSTM + EarlyStopping,LSTM,24,24,1371.594293,1966.050545,4.308456,0.906867,-7.358136
5,GRU 32 Neurons,GRU,168,24,1393.960551,1965.062114,4.404438,0.906910,133.021532
6,GRU + EarlyStopping,GRU,48,24,1378.179613,1931.757180,4.370879,0.910145,178.304365
7,GRU + EarlyStopping,GRU,24,24,1375.318934,1951.272602,4.308715,0.908262,-55.753064
8,Bi-LSTM + EarlyStopping,Bi-LSTM,48,24,1396.554288,1976.800626,4.400636,0.905846,98.702275


### Phase 6 – Selected Best Model

RNN + RMSprop (48 Hours → 24 Hours) was selected as the best-performing model after comparing all Phase 6 models across MAE, RMSE, MAPE, R², and Bias. It achieved the lowest MAE (1368.04 MW), lowest RMSE (1924.23 MW), lowest MAPE (4.3165%), highest R² (0.910843), and a very low Bias (30.37 MW), demonstrating the best overall forecasting performance among the evaluated models.

In [9]:
phase7_results = pd.read_csv("comparison/phase7_final_results.csv")
phase7_results

,Validation,MAE,RMSE,MAPE,R2,Bias
0,30 Days,1760.444645,2485.473342,5.014282,0.899805,304.760352
1,60 Days,1945.609527,2739.147156,5.341999,0.873378,282.692963
2,30 Days Peak Demand,2680.541124,3255.570997,5.509521,NaN,NaN
3,60 Days Peak Demand,3159.506809,3777.598197,6.465600,NaN,NaN
